### PR TITLE
refactor: Dataclass conversion - Access Codes 

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Part of #6465 

#### Changes proposed in this pull request:

- Data class for Access Code Models

### CODE REVIEW

File: app/models/activity.py [hunk 1]
```diff
@@ -42,22 +43,10 @@
 }
 
 
+@dataclass(init=True, repr=True, unsafe_hash=True)
```
Comment (iamareebjamal) [inline app/models/activity.py]: Why unsafe_hash?

AFAIK, init and repr are True by default as well

**Comment by iamareebjamal** [comment]
fail: GET (200) /v1/access-codes/1/marketer duration: 15017ms

File: app/models/activity.py (hunk 1 above)
Comment (mrsaicharan1) [reply app/models/activity.py]: Whenever these SQLalchemy objects are created and called, an error shows up indicating the table object to be unhashable. Unless and until it is enabled,  a __hash__() method is generated according to how eq and frozen are set.(which are not set here).

Comment (mrsaicharan1) [reply app/models/activity.py]: And yeah, `init` & `repr` are True by default.

**Comment by iamareebjamal** [comment]
```python
File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/dredd_hooks/dredd.py", line 99, in handle
    hooks._before[data.get('name')]]
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/dredd_hooks/dredd.py", line 98, in <listcomp>
    [fn(data) for fn in
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/dredd_hooks/dredd.py", line 179, in call
    result = func(*args, **kwargs)
  File "/home/travis/build/fossasia/open-event-server/tests/hook_main.py", line 265, in user_access_code
    access_code = AccessCodeFactory()
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/factory/base.py", line 46, in __call__
    return cls.create(**kwargs)
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/factory/base.py", line 564, in create
    return cls._generate(enums.CREATE_STRATEGY, kwargs)
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/factory/base.py", line 501, in _generate
    return step.build()
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/factory/builder.py", line 279, in build
    kwargs=kwargs,
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/factory/base.py", line 315, in instantiate
    return self.factory._create(model, *args, **kwargs)
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/factory/alchemy.py", line 75, in _create
    obj = model_class(*args, **kwargs)
  File "<string>", line 4, in __init__
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/sqlalchemy/orm/state.py", line 441, in _initialize_instance
    manager.dispatch.init_failure(self, args, kwargs)
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/sqlalchemy/util/langhelpers.py", line 68, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 153, in reraise
    raise value
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/sqlalchemy/orm/state.py", line 438, in _initialize_instance
    return manager.original_init(*mixed[1:], **kwargs)
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/sqlalchemy/ext/declarative/base.py", line 840, in _declarative_constructor
    "%r is an invalid keyword argument for %s" % (k, cls_.__name__)
TypeError: 'used_for' is an invalid keyword argument for AccessCode
```

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 6
           

Complexity decreasing per file
==============================
+ app/models/custom_system_role.py  -4
         

Clones removed
==============
+ app/models/custom_system_role.py  -1
+ app/models/access_code.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4220948&bid=14425796)

File: app/models/activity.py [hunk 2]
```diff
@@ -42,22 +43,14 @@
...
     __tablename__ = 'activities'
-    id = db.Column(db.Integer, primary_key=True)
-    actor = db.Column(db.String)  # user email + id
-    time = db.Column(db.DateTime(timezone=True))
-    action = db.Column(db.String)
-
-    def __init__(self, actor=None, time=None, action=None):
-        self.actor = actor
-        self.time = time
+    id: int = db.Column(db.Integer, primary_key=True)
```
Comment (niranjan94) [inline app/models/activity.py]: Codacy Issue found: [invalid syntax](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4220948)

File: app/models/custom_placeholder.py [hunk 3]
```diff
@@ -8,41 +9,22 @@ def image_default(context):
...
-                 event_sub_topic_id=None):
-        self.name = name
-        self.original_image_url = original_image_url
-        self.thumbnail_image_url = thumbnail_image_url
-        self.large_image_url = large_image_url
-        self.icon_image_url = icon_image_url
-        self.copyright = copyright
-        self.origin = origin
-        self.event_sub_topic_id = event_sub_topic_id
+    name: str = db.Column(db.String)
```
Comment (niranjan94) [inline app/models/custom_placeholder.py]: Codacy Issue found: [invalid syntax](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4220948)

File: app/models/access_code.py [hunk 4]
```diff
@@ -1,69 +1,39 @@
...
-    code = db.Column(db.String)
-    access_url = db.Column(db.String)
-    is_active = db.Column(db.Boolean)
-    tickets_number = db.Column(db.Integer)  # For event level access this holds the max. uses
-    min_quantity = db.Column(db.Integer)
-    max_quantity = db.Column(db.Integer)  # For event level access this holds the months for which it is valid
-    valid_from = db.Column(db.DateTime(timezone=True), nullable=True)
-    valid_till = db.Column(db.DateTime(timezone=True), nullable=True)
-    ticket_id = db.Column(db.Integer, db.ForeignKey('tickets.id', ondelete='CASCADE'))
+    id: int = db.Column(db.Integer, primary_key=True)
```
Comment (niranjan94) [inline app/models/access_code.py]: Codacy Issue found: [invalid syntax](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4220948)

File: app/models/custom_form_option.py [hunk 5]
```diff
@@ -1,23 +1,11 @@
...
-        self.custom_form_id = custom_form_id
-        self.value = value
-        self.deleted_at = deleted_at
-
-    def __repr__(self):
-        return '<CustomFormOption %r>' % self.id
-
-    def __str__(self):
-        return self.__repr__()
+    id: int = db.Column(db.Integer, primary_key=True)
```
Comment (niranjan94) [inline app/models/custom_form_option.py]: Codacy Issue found: [invalid syntax](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4220948)

File: app/models/custom_system_role.py [hunk 6]
```diff
@@ -1,17 +1,17 @@
...
     """Custom System Role
     """
     __tablename__ = 'custom_sys_roles'
 
-    id = db.Column(db.Integer, primary_key=True)
-    name = db.Column(db.String, unique=True)
-
-    def __init__(self, name):
-        self.name = name
+    id: int = db.Column(db.Integer, primary_key=True)
```
Comment (niranjan94) [inline app/models/custom_system_role.py]: Codacy Issue found: [invalid syntax](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4220948)

File: app/models/custom_form.py [hunk 7]
```diff
@@ -66,49 +67,23 @@
...
-        self.field_identifier = field_identifier
-        self.form = form
-        self.type = type
-        self.description = description
-        self.is_required = is_required
-        self.is_included = is_included
-        self.is_fixed = is_fixed
-        self.is_complex = is_complex
-        self.deleted_at = deleted_at
+    id: int = db.Column(db.Integer, primary_key=True)
```
Comment (niranjan94) [inline app/models/custom_form.py]: Codacy Issue found: [invalid syntax](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4220948)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones removed
==============
+ app/models/access_code.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4220948&bid=14425796)

File: app/models/access_code.py (hunk 4 above)
Comment (niranjan94) [inline app/models/access_code.py]: Codacy Issue found: [invalid syntax](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4220948)

**Comment by mrsaicharan1** [comment]
@iamareebjamal Some of the fields such as `user_id`, `used_for` were previously defined in the `__init__` methods(which are now being removed), but not mentioned in the field definitions under the `AccessCode` model. Are these being used anywhere or is it safe to remove them?

**Comment by mrsaicharan1** [comment]
> ```python
> ```python
> TypeError: 'used_for' is an invalid keyword argument for AccessCode
> ```
> 
> 
>       
> ```


`TypeError: 'used_for' is an invalid keyword argument for AccessCode`

**Comment by iamareebjamal** [comment]
It's fine to remove. Just see no one is using user_id as keyword argument

**Comment by mrsaicharan1** [comment]
@iamareebjamal 

> fail: GET (200) /v1/access-codes/1/marketer duration: 15017ms

This was failing because there was no marketer id in the factory definition. I think this is GTG now.

**Pull request merged**